src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs

```diff
@@ -289,6 +289,11 @@ public static IResourceBuilder<AzurePostgresFlexibleServerResource> RunAsContain
         }
 
         var azureResource = builder.Resource;
+        var azureDatabases = builder.ApplicationBuilder.Resources
+            .OfType<AzurePostgresFlexibleServerDatabaseResource>()
+            .Where(db => db.Parent == azureResource)
+            .ToDictionary(db => db.Name);
+
         RemoveAzureResources(builder.ApplicationBuilder, azureResource);
```

Comment:
Not a big issue, but this results in scanning the resource collection twice to get the same set of data from above. Why not just iterate over the resources inline or do the deletes in the loop below.

Reply:
I can pass the dictionary into the Remove method.